server/geoip_service.py

```diff
@@ -69,23 +70,45 @@ def __init__(self):
             Download the geoip database to a file. If the downloaded file is not
         a valid gzip file, then it does NOT overwrite the old file.
         """
+        if not config.GEO_IP_LICENSE_KEY:
+            self._logger.warning(
+                "GEO_IP_LICENSE_KEY not set! Unable to download GeoIP database!"
+            )
+            return
 
         self._logger.info("Downloading new geoip database")
 
         # Download new file to a temp location
-        temp_file_path = "/tmp/geoip.mmdb.gz"
-        await self._download_file(config.GEO_IP_DATABASE_URL, temp_file_path)
+        temp_file_path = "/tmp/geoip.mmdb.tar.gz"
+        await self._download_file(
+            config.GEO_IP_DATABASE_URL,
+            config.GEO_IP_LICENSE_KEY,
+            temp_file_path
+        )
+
+        def get_db_member(tar: tarfile.TarFile) -> str:
+            for name in tar.getnames():
+                if name.endswith("GeoLite2-Country.mmdb"):
+                    return name
+
+            # Because we verified the checksum earlier, this should only be
+            # possible if maxmind actually served us a bad file
+            raise Exception("Tar archive did not contain the database file!")
 
         # Unzip the archive and overwrite the old file
         try:
-            with gzip.open(temp_file_path, 'rb') as f_in:
+            with tarfile.open(temp_file_path, 'r:gz') as tar:
+                name = get_db_member(tar)
                 with open(self.file_path, 'wb') as f_out:
+                    f_in = tar.extractfile(name)
+                    assert f_in is not None
```

Comment:
what's the reason behind of it?

Reply:
Like, you `get_db_member` raising if there is no file, isn't it?

Reply:
Because `tar.extractfile` has return type `Optional[IO[bytes]]` I get a type error (warning) if I use it directly since the type checker doesn't know about the relationship between `tar.getnames` and `tar.extractfile`. 

The assert should theoretically always succeed (unless there is a bug in `tarfile` I guess) but it makes the type checker happy.

Reply:
=^_^= couldn't see that without analyzer. Yes, i thought about wrong tar too, but i don't know, if there is such possibility, when we already could read name.. like it's highly non possible to not to have whole file, isn't it? :-)

Reply:
Yea, like I said. I don't think it's possible for that assert to fail.

Reply:
Unless someone's feeling malicious and the tar has a directory with `GeoLite2-Country.mmdb` in it. IMHO it does not hurt to just throw.